### ISSUE TITLE
nested-structs false positives with "struct{}"

### ISSUE THREAD

**Issue by sbunce** [mgechev/revive#824]
I found some instances of "struct{}" causing the nested-structs check to fail. Based on other issues it looks like the intention is to not include struct{} in this check.
```
type test struct {
  foo []chan struct{}
  bar map[string]struct{}
}
```

If this is working as intended feel free to resolve.

p.s. Revive works great! Found multiple bugs in my code.

**Comment by mgechev** [comment]
Thank you!

Can't reproduce it in https://github.com/mgechev/revive/pull/826. Would you share another example which gives you the false positive?

**Comment by sbunce** [comment]
I created a draft pull request which shows the issue. https://github.com/mgechev/revive/pull/828

I think it's an issue in the "nested-structs" rule.

Running "make test" shows the following.
```
    utils.go:120: Unexpected problem at nested-structs.go:50: no nested structs are allowed
    utils.go:120: Unexpected problem at nested-structs.go:51: no nested structs are allowed
--- FAIL: TestNestedStructs (0.00s)
```

**Comment by chavacava** [comment]
Hi @sbunce, thanks for the PR with the counter example. I'll check for a solution.

**Comment by chavacava** [comment]
@sbunce please let me know if the fix works for your cases. If that is the case, you can close the PR #828 
Thanks again!

**Comment by sbunce** [comment]
Fix works. Thank you!

I found one more tiny one with generics.
```
type test struct {
	baz generic[struct{}]   // Must not match
}

type generic [T any] struct {}
```

The error points at the "baz" field.
```
=== RUN   TestNestedStructs
    utils.go:119: Unexpected problem at nested-structs.go:64: no nested structs are allowed
--- FAIL: TestNestedStructs (0.00s)
```

I added the example to my draft PR. https://github.com/mgechev/revive/pull/828/files
